[32mINFO    [0m integration.ha_tests.test_upgrade_from_stable:test_upgrade_from_stable.py:102 starting continuous writes to the database
[33mWARNING [0m juju.model:model.py:1563 relate is deprecated and will be removed. Use integrate instead.
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  postgresql/0 [executing] active: Primary
  postgresql/1 [executing] active: 
  postgresql/2 [executing] active: 
  postgresql-test-app/0 [idle] active: 
[32mINFO    [0m integration.ha_tests.test_upgrade_from_stable:test_upgrade_from_stable.py:106 checking whether writes are increasing
[32mINFO    [0m integration.ha_tests.test_upgrade_from_stable:test_upgrade_from_stable.py:115 Build charm locally
[32mINFO    [0m integration.ha_tests.test_upgrade_from_stable:test_upgrade_from_stable.py:118 Refresh the charm
[32mINFO    [0m integration.ha_tests.test_upgrade_from_stable:test_upgrade_from_stable.py:121 Wait for upgrade to start
[32mINFO    [0m integration.ha_tests.test_upgrade_from_stable:test_upgrade_from_stable.py:128 Wait for upgrade to complete
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  postgresql/0 [idle] active: Primary
  postgresql/1 [executing] active: 
  postgresql/2 [executing] waiting: other units upgrading first...
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  postgresql/0 [idle] waiting: other units upgrading first...
  postgresql/1 [idle] waiting: other units upgrading first...
  postgresql/2 [executing] maintenance: refreshing the snap
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  postgresql/0 [executing] waiting: other units upgrading first...
  postgresql/1 [idle] waiting: other units upgrading first...
  postgresql/2 [idle] maintenance: upgrade completed
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  postgresql/0 [idle] waiting: other units upgrading first...
  postgresql/1 [executing] maintenance: refreshing the snap
  postgresql/2 [idle] maintenance: upgrade completed
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  postgresql/0 [idle] active: 
  postgresql/1 [executing] maintenance: refreshing the snap
  postgresql/2 [idle] maintenance: upgrade completed
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  postgresql/0 [executing] maintenance: updating configuration
  postgresql/1 [idle] maintenance: upgrade completed
  postgresql/2 [idle] maintenance: upgrade completed
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  postgresql/0 [executing] maintenance: refreshing the snap
  postgresql/1 [idle] maintenance: upgrade completed
  postgresql/2 [idle] maintenance: upgrade completed
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  postgresql/0 [executing] active: 
  postgresql/1 [executing] active: 
  postgresql/2 [executing] active: Primary
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  postgresql/0 [executing] active: 
  postgresql/1 [idle] active: 
  postgresql/2 [idle] active: Primary
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  postgresql/0 [idle] active: 
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  postgresql/0 [idle] active: 
  postgresql/1 [idle] active: 
  postgresql/2 [idle] active: Primary
[32mINFO    [0m integration.ha_tests.test_upgrade_from_stable:test_upgrade_from_stable.py:135 checking whether writes are increasing
[32mINFO    [0m integration.ha_tests.test_upgrade_from_stable:test_upgrade_from_stable.py:140 checking whether no writes were lost
[32mINFO    [0m integration.ha_tests.test_upgrade_from_stable:test_upgrade_from_stable.py:145 checking the number of switchovers
[32mINFO    [0m pytest_operator.plugin:plugin.py:903 Model status:

Model  Controller           Cloud/Region         Version  SLA          Timestamp
test   localhost-localhost  localhost/localhost  3.4.6    unsupported  01:48:09Z

App                  Version  Status  Scale  Charm                Channel      Rev  Exposed  Message
postgresql           14.13    active      3  postgresql                          0  no       
postgresql-test-app           active      1  postgresql-test-app  latest/edge  279  no       received database credentials of the first database

Unit                    Workload  Agent  Machine  Public address  Ports     Message
postgresql-test-app/0*  active    idle   3        10.39.193.147             received database credentials of the first database
postgresql/0*           active    idle   0        10.39.193.9     5432/tcp  
postgresql/1            active    idle   1        10.39.193.50    5432/tcp  
postgresql/2            active    idle   2        10.39.193.122   5432/tcp  Primary

Machine  State    Address        Inst id        Base          AZ  Message
0        started  10.39.193.9    juju-deeeac-0  ubuntu@22.04      Running
1        started  10.39.193.50   juju-deeeac-1  ubuntu@22.04      Running
2        started  10.39.193.122  juju-deeeac-2  ubuntu@22.04      Running
3        started  10.39.193.147  juju-deeeac-3  ubuntu@22.04      Running

[32mINFO    [0m pytest_operator.plugin:plugin.py:909 Juju error logs:

machine-0: 01:34:20 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-0: 01:34:20 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-0: 01:34:21 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-1: 01:34:26 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-1: 01:34:26 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-1: 01:34:26 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-3: 01:34:30 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-3: 01:34:30 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-test-app-0: 01:34:31 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-2: 01:34:33 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-2: 01:34:33 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-2: 01:34:33 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-postgresql-0: 01:35:41 ERROR unit.postgresql/0.juju-log Failed to list PostgreSQL database users: could not translate host name "None" to address: Temporary failure in name resolution

unit-postgresql-0: 01:46:26 ERROR unit.postgresql/0.juju-log upgrade:1: failed to change plugins: 
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-0/charm/lib/charms/postgresql_k8s/v0/postgresql.py", line 346, in enable_disable_extensions
    cursor.execute(
psycopg2.errors.ObjectNotInPrerequisiteState: pgaudit must be loaded via shared_preload_libraries


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-0/charm/./src/charm.py", line 1058, in enable_disable_extensions
    self.postgresql.enable_disable_extensions(extensions, database)
  File "/var/lib/juju/agents/unit-postgresql-0/charm/lib/charms/tempo_coordinator_k8s/v0/charm_tracing.py", line 735, in wrapped_function
    return callable(*args, **kwargs)  # type: ignore
  File "/var/lib/juju/agents/unit-postgresql-0/charm/lib/charms/postgresql_k8s/v0/postgresql.py", line 357, in enable_disable_extensions
    raise PostgreSQLEnableDisableExtensionError()
charms.postgresql_k8s.v0.postgresql.PostgreSQLEnableDisableExtensionError
unit-postgresql-test-app-0: 01:47:28 ERROR unit.postgresql-test-app/0.juju-log database:4: Unable to read result
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 434, in _stop_continuous_writes
    with attempt, open(LAST_WRITTEN_FILE) as fd:
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/last_written_value'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 433, in _stop_continuous_writes
    for attempt in Retrying(stop=stop_after_delay(60), wait=wait_fixed(5)):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/venv/tenacity/__init__.py", line 443, in __iter__
    do = self.iter(retry_state=retry_state)
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/venv/tenacity/__init__.py", line 376, in iter
    result = action(retry_state)
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/venv/tenacity/__init__.py", line 419, in exc_check
    raise retry_exc from fut.exception()
tenacity.RetryError: RetryError[<Future at 0x7fba181c9ab0 state=finished raised FileNotFoundError>]

[32mINFO    [0m pytest_operator.plugin:plugin.py:991 Forgetting model main...